### PR TITLE
Introduced 2d arrays in Variant

### PR DESCRIPTION
**Pull request by aalkin**
* added matrix<T> struct
* added parsers for matrix into/from boost ptree and string representation
* extended Variant to use matrix<int/float/double>
* added matrix data memmbers handling for serialization
* updated configurableObjects.cxx example

### CODE REVIEW

**Comment by ktf** [comment]
I would prefer a name like `Array2D`, rather than `Matrix`. Any particular reason for your choice?

**Comment by aalkin** [comment]
> I would prefer a name like `Array2D`, rather than `Matrix`. Any particular reason for your choice?

No reason, just the first name that came to mind. I'll rename to `Array2D` when the tests finish.

**Review by ktf — DISMISSED** [review]

File: Framework/Core/include/Framework/ConfigParamRegistry.h [hunk 1]
```diff
@@ -80,6 +103,8 @@ class ConfigParamRegistry
         return std::string_view{mStore->store().get<std::string>(key)};
       } else if constexpr (is_base_of_template<std::vector, T>::value) {
         return extractVector<typename T::value_type>(mStore->store().get_child(key));
+      } else if constexpr (is_base_of_template<o2::framework::Array2D, T>::value) {
+        return extractMatrix<typename T::element_t>(mStore->store().get_child(key));
```
Comment (ktf) [inline Framework/Core/include/Framework/ConfigParamRegistry.h:107]: should probably be `extractArray2D`, fine for now.

File: Framework/Core/include/Framework/ConfigParamsHelper.h [hunk 2]
```diff
@@ -119,7 +119,10 @@ struct ConfigParamsHelper {
                          V == VariantType::ArrayFloat ||
                          V == VariantType::ArrayDouble ||
                          V == VariantType::ArrayBool ||
-                         V == VariantType::ArrayString) {
+                         V == VariantType::ArrayString ||
+                         V == VariantType::MatrixInt ||
+                         V == VariantType::MatrixFloat ||
+                         V == VariantType::MatrixDouble) {
```
Comment (ktf) [inline Framework/Core/include/Framework/ConfigParamsHelper.h:125]: Should probably be Array2DInt, etc.

**Comment by ktf** [comment]
Merging this, however please make sure you consistently use Array2D everywhere.

**Pull request merged**